repository/src/main/java/org/alfresco/repo/action/ActionServiceImpl.java

```diff
@@ -588,11 +593,24 @@ public void executeAction(Action action, NodeRef actionedUponNodeRef, boolean ch
         }
         else
         {
+            verifyActionAccessRestrictions(action);

```

Comment:
Added verification here for async action executions to check restrictions immediatelly. For async actions they will run when they are created and when they are executed, a bit of redundancy, but didn't see a feasible way to avoid it.